### DISCUSSION TITLE
SSL/certificates on QNAP native package.

### DISCUSSION THREAD

**Discussion by SageCloud2** [nzbgetcom/nzbget#377] [Q&A] [answered]
I've installed the QNAP package and want to use SSL encrypted transport when accessing the WebUI. All is configured for ports etc., except for the SSL cert and chain. The error message which keeps on appearing: 
`Could not initialize TLS, secure certificate is not configured or the cert-file was not found. Check option <SecureCert>`

QNAP is now enabling users to integrate Let'sEncrypt certificates in their OS which renews automatically. The drawback is that it is unknown where to find the required files for NZBGet. 
I have found that I need to configure <SecureCert> and <SecureKey> but the information provide tells me to map to fullchain.pem and privkey.pem. These files do not exist, only 'cert' and 'key' in /etc/config/QcloudSSLCertificate/cert which seem to be the files needed. I am unsure how to continue.

**Comment by phnzb** [comment, marked as answer]
@SageCloud2 
In my setup (QNAP TS-253E / QTS 5.1.7.2770) I have found certificate files in /etc/stunnel/. I tested and SSL seems to work with the following settings:

`SecureControl`: Yes
`SecurePort`: 6791
`SecureCert`: /etc/stunnel/backup.cert
`SecureKey`: /etc/stunnel/backup.key

Please note - HTTP can't be disabled

**Reply by SageCloud2** [reply]
Thank you! This indeed seems to do the job. 
Thank you again; it solved an issue I've been looking at for days already.
